Hi,

I am the current maintainer of libdivide. I have now fixed the ```TRY_RUN()``` cross compilation error (at least it works for me when cross compiling using MinGW on x64).

The issue below however seems to be a x86 MSVC (32-bit) bug that is caused by libdivide's AVX2 code.

```
C:\vsts\_work\2\s\buildtrees\libdivide\src\v2.0-66dc5347d1\libdivide.h(1570) : fatal error C1001: An internal error has occurred in the compiler.
(compiler file 'd:\agent\_work\3\s\src\vctools\Compiler\Utc\src\p2\main.c', line 160)
 To work around this problem, try simplifying or changing the program near the locations listed above.
Please choose the Technical Support command on the Visual C++ 
 Help menu, or open the Technical Support help file for more information

INTERNAL COMPILER ERROR in 'C:\VS2017\Stable\VC\Tools\MSVC\14.22.27905\bin\Hostx64\x86\cl.exe'
    Please choose the Technical Support command on the Visual C++
    Help menu, or open the Technical Support help file for more information
```

Since I don't know how long it will take until this bug gets fixed in MSVC I have added an option to libdivide's CMake build system which allows to turn off CPU vector instructions (when building the test programs). So the bug above can now be avoided using e.g.:

```bash
cmake .. -G "Visual Studio 15 2017" -DENABLE_VECTOR_EXTENSIONS=OFF
cmake --build . --config Release
```

I have also added ```find_package(libdivide)``` support to libdivide's ```CMakeLists.txt``` and made a bunch of other changes that should make it easier to package libdivide. I will do a new libdivide release once I get confirmation that the new CMake code works well with vcpkg.

--- a/ports/libdivide/CONTROL
+++ b/ports/libdivide/CONTROL
@@ -1,4 +1,7 @@
 Source: libdivide
-Version: 2.0
+Version: 3.0
 Homepage: https://github.com/ridiculousfish/libdivide
 Description: libdivide.h is a header-only C/C++ library for optimizing integer division.
+
+Feature: test
+Description: Build test

--- a/ports/libdivide/portfile.cmake
+++ b/ports/libdivide/portfile.cmake
@@ -1,25 +1,31 @@
-include(vcpkg_common_functions)
-
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO ridiculousfish/libdivide
-    REF v2.0
-    SHA512 0599c6d6206d8a7273804f65e79c32df47b0e7de9703460201c2eb8a480542d88ad6d5d8b8135d576805892edfd88ef304249f51667088c8f8a9db4c32cbb9e2
+    REF v3.0
+    SHA512 fae17a4125c3b17aeb37283d7bba9fea2e4d3b208861d6ed81a6cdcf5dbf3286cf676cedba99c73a16115cf8bf9dcbd2cf6a48ca52fb85d4b0b24024e53d055e
     HEAD_REF master
+)
+
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    test BUILD_TESTS
 )
 
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
     PREFER_NINJA 
-    OPTIONS
+    OPTIONS ${FEATURE_OPTIONS}
       -DLIBDIVIDE_SSE2=OFF
       -DLIBDIVIDE_AVX2=OFF
       -DLIBDIVIDE_AVX512=OFF
+      -DENABLE_VECTOR_EXTENSIONS=OFF
 )
 
 vcpkg_install_cmake()
 
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/${PORT})
+
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/lib)
 
 # Handle copyright
-file(INSTALL ${SOURCE_PATH}/LICENSE.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+file(INSTALL ${SOURCE_PATH}/LICENSE.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright) 

